projects/swimlane/ngx-datatable/src/lib/components/header/header.component.ts

```diff
@@ -58,25 +67,25 @@ import { MouseEvent } from '../../events';
   changeDetection: ChangeDetectionStrategy.OnPush
 })
 export class DataTableHeaderComponent {
-
   @Input() sortAscendingIcon: any;
   @Input() sortDescendingIcon: any;
   @Input() scrollbarH: boolean;
   @Input() dealsWithGroup: boolean;
   @Input() targetMarkerTemplate: any;
-  
+
   targetMarkerContext: any;
 
   @Input() set innerWidth(val: number) {
     this._innerWidth = val;
-
-    if (this._columns) {    
-      const colByPin = columnsByPin(this._columns);
-      this._columnGroupWidths = columnGroupWidths(colByPin, this._columns);
-      this.setStylesByGroup();
-    }
+    setTimeout(() => {
+      if (this._columns) {
```

Comment:
can you skip the setTimeout by moving the if check outside of it? or is the timeout to allow `this._columns` to populate?

Reply:
Yup, it basically skips the cycle allowing _columns to populate. This is a change in behavior caused by Ivy.